### ISSUE TITLE
[bug] Cannot add property _currentElement, object is not extensible

### ISSUE THREAD

**Issue by boopathi** [jsdom/jsdom#3175]
### Basic info:

- **Node.js version:** 12, 14
- **jsdom version:** 16.5.3

### Minimal reproduction case

I pushed a minimal reproduction to this repo - https://github.com/boopathi/jsdom-bug-1 

The CI build in that repo would provide the error details - https://github.com/boopathi/jsdom-bug-1/runs/2328255180?check_suite_focus=true

```js
const { PurgeCSS } = require("purgecss");
const fromHtml = require("purgecss-from-html");

// no need to use this. using purgecss-from-html which uses parse5 would fail
require("jsdom");

// use purgecss and fromHtml
// purgecss would fail because JSDOM extends an internal parse5 method.
```

When JSDOM and parse5 (purgecss-from-html uses parse5) are used imported in the same context, the following error is thrown - 

```
TypeError: Cannot add property _currentElement, object is not extensible
```

The stack trace of this error points to https://github.com/jsdom/jsdom/blob/e4c4004c122797cd897f35475bb6ab42688ba452/lib/jsdom/browser/parser/html.js#L21-L34

### How does similar code behave in browsers?

This section is not relevant for this issue.

(Link to a jsbin or similar strongly suggested.)

**Comment by domenic** [comment]
Yeah, jsdom is not compatible with other things which monkeypatch parse5, I guess. It's a shame we have to do so, but until https://github.com/inikulin/parse5/issues/237 is fixed I don't see a way around that.

**Comment by TimothyGu** [comment]
Could we perhaps be at least a bit more careful with the monkey patching? I.e., do something like

```diff
 OpenElementStack.prototype.push = function (...args) {
   openElementStackOriginalPush.apply(this, args);
 
+  if (this.treeAdapter instanceof JSDOMParse5Adapter) {
     this.treeAdapter._currentElement = this.current;
+  }
 
   const after = this.items[this.stackTop];
   if (after._pushedOnStackOfOpenElements) {
     after._pushedOnStackOfOpenElements();
   }
 };
```

That'll fix this case at least.

**Comment by bilal68** [comment]
how can I slect all the img tags with the property "itemId" present in them i.e:
`<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=us-ascii">
</head>
<body>
<img src="" itemId="someId"></img>
<img src="" itemId="someId"></img>
<img src="" itemId="someId"></img>
</body>
</html>
`

**Comment by domenic** [comment]
This is almost certainly fixed by https://github.com/jsdom/jsdom/pull/3354. I won't add a regression test since pulling in third-party dependencies just for testing is a bit annoying.
